main.go

```diff
@@ -38,20 +39,32 @@ var glcm = common.GetLifecycleMgr()
 func main() {
 	pipeline.SetLogSanitizer(common.NewAzCopyLogSanitizer()) // make sure ForceLog logs get secrets redacted
 
+	// note: azcopyAppPathFolder is the default location for all AzCopy data (logs, job plans, oauth token on Windows)
+	// but both logs and job plans can be put elsewhere as they can become very large
 	azcopyAppPathFolder := GetAzCopyAppPath()
+
+	// the user can optionally put the log files somewhere else
 	azcopyLogPathFolder := common.GetLifecycleMgr().GetEnvironmentVariable(common.EEnvironmentVariable.LogLocation())
 	if azcopyLogPathFolder == "" {
 		azcopyLogPathFolder = azcopyAppPathFolder
 	}
 
+	// the user can optionally put the plan files somewhere else
+	azcopyJobPlanFolder := common.GetLifecycleMgr().GetEnvironmentVariable(common.EEnvironmentVariable.JobPlanLocation())
+	if azcopyJobPlanFolder == "" {
+		azcopyJobPlanFolder = path.Join(azcopyAppPathFolder, "plans")
+	}
+	if err := os.Mkdir(azcopyJobPlanFolder, os.ModeDir|os.ModePerm); err != nil && !os.IsExist(err) {
```

Comment:
We force creation of the plan dir, but I'm not sure whether we force creation of the log dir. (In the old code, forcing the creation of the plan dir automatically forced creation of log dir, because plan was child of log. But that's not the case now).

Do we need to add code to create the log dir too, now that it might not be parent of the plan dir?

Reply:
Good point, we should be consistent and force create both locations